From f23cc6f8e7d224ee74a4f9d67b91587343e6de60 Mon Sep 17 00:00:00 2001
From: Thomas Schoebel-Theuer <tst@1und1.de>
Date: Fri, 28 Jun 2013 11:03:14 +0200
Subject: [PATCH] mars: SPECIAL for in-tree build

---
 block/Kconfig.v5.4  | 2 ++
 block/Makefile | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/block/Kconfig b/block/Kconfig
index 41c0917ce622..ccf181798cb7 100644
--- a/block/Kconfig
+++ b/block/Kconfig
@@ -183,6 +183,8 @@ source "block/partitions/Kconfig.v5.4"
 
 endmenu
 
+source "block/mars/kernel/Kconfig.v5.4"
+
 endif # BLOCK
 
 config BLOCK_COMPAT
diff --git a/block/Makefile b/block/Makefile
index 9ef57ace90d4..c074adff6d3a 100644
--- a/block/Makefile
+++ b/block/Makefile
@@ -36,3 +36,5 @@ obj-$(CONFIG_BLK_DEBUG_FS)	+= blk-mq-debugfs.o
 obj-$(CONFIG_BLK_DEBUG_FS_ZONED)+= blk-mq-debugfs-zoned.o
 obj-$(CONFIG_BLK_SED_OPAL)	+= sed-opal.o
 obj-$(CONFIG_BLK_PM)		+= blk-pm.o
+
+obj-$(CONFIG_MARS)		+= mars/kernel/
-- 
2.26.2

